examples/ecies-example.cpp

```diff
@@ -0,0 +1,148 @@
+static const auto macFunc = std::function<std::unique_ptr<MessageAuthenticationCode>(const std::vector<uint8_t>&)>(
+            [](const std::vector<uint8_t> &key) -> std::unique_ptr<MessageAuthenticationCode> {
+    return std::make_unique<mococrw::HMAC>(digest, key);
+});
+
+std::unique_ptr<ECIESEncryptionCtx> buildEncryptionContext()
+{
+    ECIESCtxBuilder encBuilder;
+    auto pubKey = AsymmetricPublicKey::readPublicKeyFromPEM(pubKeyPem);
+
+    return encBuilder.setKDF(kdfFunc)
+            .setMacFactoryFunction(macFunc)
+            .setMacKeySize(digestLength)
+            .buildEncryptionCtx(pubKey);
+}
+
+std::unique_ptr<ECIESDecryptionCtx> buildDecryptionContext(const std::vector<uint8_t> &ephKey)
+{
+    ECIESCtxBuilder decBuilder;
+    auto privKey = AsymmetricPrivateKey::readPrivateKeyFromPEM(privKeyPem, "");
+    std::shared_ptr<AsymmetricKey::Spec> spec = privKey.getKeySpec();
+    auto eccSpec = std::dynamic_pointer_cast<ECCSpec>(spec);
+    if (!eccSpec) {
+        std::cerr << "Given key is no ECC key." << std::endl;
+        exit(EXIT_FAILURE);
+    }
+
+    auto _ephKey = AsymmetricPublicKey::fromECPoint(eccSpec, ephKey);
+
+    return decBuilder.setKDF(kdfFunc)
+            .setMacFactoryFunction(macFunc)
+            .setMacKeySize(digestLength)
+            .buildDecryptionCtx(privKey, _ephKey);
+}
+
+struct EciesEncryptResult encrypt_ecies()
+{
+    std::unique_ptr<ECIESEncryptionCtx> encCtx = buildEncryptionContext();
+    std::vector<uint8_t> message = utility::fromHex("deadbeef");
+
+    std::vector<uint8_t> encryptedData;
+    try {
+         encCtx->update(message);
+         encryptedData = encCtx->finish();
+    }  catch (MoCOCrWException &e) {
+        std::cerr << "Error encrypting data." << std::endl;
+        std::cerr << e.what() << std::endl;
+        exit(EXIT_FAILURE);
+    }
+
+    return EciesEncryptResult{
+        encCtx->getEphemeralKey().toECPoint(openssl::EllipticCurvePointConversionForm::uncompressed),
+        encryptedData,
+        encCtx->getMAC(),
+    };
+}
+
+void decrypt_ecies(struct EciesEncryptResult eciesData)
+{
+    /* We need
+     * - mac value
+     * - ephemeral key
+     * - private key
+     */
+
+    /* Use the default values, which should match here. */
+    auto decCtx = buildDecryptionContext(eciesData.ephKey);
+    std::vector<uint8_t> decryptedData;
+    try {
+        /* Decrypt the ciphertext */
+        decCtx->update(eciesData.ciphertext);
+
+        /* Set the received mac value */
+        decCtx->setMAC(eciesData.mac);
+
+        /* Get the plaintext and verify the MAC */
+        decryptedData = decCtx->finish();
+    } catch (MoCOCrWException &e) {
+        std::cerr << "Error decrypting integrated encryption scheme." << std::endl;
+        std::cerr << e.what();
+        exit(EXIT_FAILURE);
+    }
+}
+
+int main(void)
+{
+    auto eciesData = encrypt_ecies();
+    decrypt_ecies(eciesData);
```

Comment:
In the AES example, you compared plaintext to decrypted ciphertext. Do you also want to do this here?

Reply:
It's the same as for the AEAD encryption. If the MAC verification passes, the decrypted ciphertext should be identical to the plaintext. This is checked in the unit tests.
But if this help the user of this example, I can add it. Please leave a short comment :)

Reply:
Same as above, fine with me to not check this explicitly.

Reply:
I guess this is still open?